pilot/pkg/serviceregistry/kube/controller/ambient/services.go

```diff
@@ -193,6 +193,17 @@ func (a *index) constructService(svc *v1.Service, w *Waypoint) *workloadapi.Serv
 			Mode: workloadapi.LoadBalancing_STRICT,
 		}
 	}
+	ipPolicy := workloadapi.IPFamilies_AUTOMATIC
+	if len(svc.Spec.IPFamilies) == 2 {
```

Comment:
This seems potentially a little brittle, can users do silly things like put v1.IPv4Protocol in the list twice ?

Reply:
No the validation is pretty strict. We could make it more robust anyways though